Chessington.GameEngine/Pieces/King.cs

```diff
@@ -10,7 +10,16 @@ public King(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var currentSquare = board.FindPiece(this);
+
+            var minusOneToOne = new List<int>(new[] { -1, 0, 1 });
+
+            var moves = minusOneToOne.SelectMany(
+                x => minusOneToOne,
+                (rowChange, colChange) => new MoveVector(rowChange, colChange)
+            );
```

Comment:
As it's only 8 moves (unless I'm misremembering the chess rules?) I would have probably just written them out manually as a list of `MoveVector`s, as that's quite a bit more readable in my opinion 🙂

Reply:
Yeah that's definitely a more readable way of doing it 🙂 Changed in d4877e8